docs/project/development-guidelines.md

```diff
@@ -0,0 +1,238 @@
+
+**Wrong:**
+
+By creating the variable this.conn and receiving the connection, we have already passed it to the mother class's knex.
+
+``` jsx
+// bad
+
+class UserRepository extends Repository {
+    constructor(injection) {
+        super({
+            entity: User,
+            table: "users",
+            knex: connection
+        })
+
+        this.conn = connection
+    }
+
+    async getUserByNickName(nickname) {
+        const ret = await this.conn('users').select('*')
+        .where('nickname', nickname)
+       
+        if (!ret[0] || ret[0].length == 0) return undefined
+        return User.fromJSON(ret[0])
+    }
+}
+
+```
+
+**Correct:**
+
+Call **this.knex** directly in our methods
+
+``` jsx
+// well
+
+class UserRepository extends Repository {
+    constructor(injection) {
+        super({
+            entity: User,
+            table: "users",
+            knex: connection
+        })
+    }
+
+    async getUserByNickName(nickname) {
+        const ret = await this.**knex**('users').select('*')
+         .where('nickname', nickname)
+
+        if (!ret[0] || ret[0].length == 0) return undefined
+        return User.fromJSON(ret[0])
+    }
+}
+
+```
+
+## Handling usecase failures
+
+To carry out the treatment of errors that occurred during the execution of the usecase, enable the environment variable HERBS_EXCEPTION in the production environments so that Herbs can obfuscate the errors generated and handle it on its return.
+
+``` jsx
+// bad
+const createProduct = injection =>
+    usecase('Create Product', {
+
+        request: {
+            name: String,
+            ...
+        },
+
+        response: {
+            product: Product
+        },
+
+        'Check if the Product is valid': step(ctx => {
+           try {
+                if (!isValid) return Err(errors)
+                return Ok()
+            }
+            catch(e) {
+                throw new "Error to validated product"
+            }
+        }),
+
+        'Save the Product to the repository': step(async ctx => {
+            try {
+                ctx.ret.product = await repo.insert(product)
+                return Ok()
+            }
+            catch(e) {
+                throw new "Error to save product"
+            }
+        }),
+```
+
+``` jsx
+// well
+
+HERBS_EXCEPTION = audit
+
+const createProduct = injection =>
+    usecase('Create Product', {
+
+        request: {
+            name: String,
+            ...
+        },
+
+        response: {
+            product: Product
+        },
+
+        'Check if the Product is valid': step(ctx => {
+            if (!isValid) return Err(errors)
+            return Ok()
+        }),
+
+        'Save the Product to the repository': step(async ctx => {
+            ctx.ret.product = await repo.insert(product)
+            return Ok()
+        }),
+```
+
+
+## Creating the business rules workflow
```

Comment:
## Make your business workflow more explicit

You can use `IfElse` to better document your business workflow. This makes your code easier to understand and maintain.

--

Also link to the `IfElse` docs...